src/status_im/chat/models/message.cljs

```diff
@@ -136,7 +136,9 @@
                                      (not= from current-public-key))
                                 (update-in [:chats chat-id :loaded-unviewed-messages-ids]
                                            (fnil conj #{}) message-id))
-               :data-store/tx [(messages-store/save-message-tx prepared-message)]}
+               :data-store/tx [(messages-store/save-message-tx
+                                prepared-message
+                                #(re-frame/dispatch [:message/message-persisted raw-message]))]}
```

Comment:
Dispatching an event per each received message may have a negative effect on performance if a batch of them was received. This can be optimized later.

Reply:
I don’t see how it is much worse than it was, taking into account that we open a tx per message and always were opening it. But sure, the room for optimization is there.

Reply:
We do not open tx per message: all `:data-store/tx` effects which are produced during a single event handling are accumulated and executed in a single transaction. See the list of effects which are "merged" this way here https://github.com/status-im/status-react/blob/1104becfbab11506b98028ce81ad6ff869381f51/src/status_im/utils/fx.cljs#L13